controllers/hostingdeploymentautoscalingjob/hostingdeploymentautoscalingjob_controller_test.go

```diff
@@ -0,0 +1,762 @@
+				AddRegisterScalableTargetsResponse(applicationautoscaling.RegisterScalableTargetOutput{}).
+				AddPutScalingPolicyErrorResponse("InternalServiceException", "Server Error", 500, "request id").
+				Build()
+			controller = createReconciler(k8sClient, applicationAutoscalingClient)
+			request = CreateReconciliationRequest(hostingdeploymentautoscalingjob.ObjectMeta.Name, hostingdeploymentautoscalingjob.ObjectMeta.Namespace)
+
+		})
+
+		AfterEach(func() {
+			Expect(receivedRequests.Len()).To(Equal(6))
+		})
+
+		It("should requeue immediately", func() {
+			result, err := controller.Reconcile(request)
+
+			Expect(err).ToNot(HaveOccurred())
+			// TODO mbaijal: Check this
+			//Expect(result.Requeue).To(Equal(true))
+			Expect(result.RequeueAfter).To(Equal(time.Duration(0)))
+		})
+	})
+
+	Context("When the second describe fails with a server error", func() {
+		BeforeEach(func() {
+			receivedRequests = List{}
+			mockAutoscalingClientBuilder = NewMockAutoscalingClientBuilder(GinkgoT()).WithRequestList(&receivedRequests)
+			applicationAutoscalingClient := mockAutoscalingClientBuilder.
+				AddDescribeScalableTargetsResponse(outOfDateTargetDescription).
+				AddDescribeScalingPoliciesResponse(outOfDatePolicyDescription).
+				AddDeleteScalingPolicyResponse(applicationautoscaling.DeleteScalingPolicyOutput{}).
+				AddDeregisterScalableTargetsResponse(applicationautoscaling.DeregisterScalableTargetOutput{}).
+				AddRegisterScalableTargetsResponse(applicationautoscaling.RegisterScalableTargetOutput{}).
+				AddPutScalingPolicyResponse(applicationautoscaling.PutScalingPolicyOutput{}).
+				AddDescribeScalableTargetsResponse(applicationautoscaling.DescribeScalableTargetsOutput{}).
+				AddDescribeScalingPoliciesErrorResponse("InternalServiceException", "Server Error", 500, "request-id").
+				Build()
+			controller = createReconciler(k8sClient, applicationAutoscalingClient)
+			request = CreateReconciliationRequest(hostingdeploymentautoscalingjob.ObjectMeta.Name, hostingdeploymentautoscalingjob.ObjectMeta.Namespace)
+
+		})
+
+		AfterEach(func() {
+			Expect(receivedRequests.Len()).To(Equal(8))
+		})
+
+		It("should requeue immediately", func() {
+			result, err := controller.Reconcile(request)
+
+			Expect(err).ToNot(HaveOccurred())
+			// TODO mbaijal: Check this
+			//Expect(result.Requeue).To(Equal(true))
+			Expect(result.RequeueAfter).To(Equal(time.Duration(0)))
+		})
+
+		It("should update the status with the error message", func() {
+			controller.Reconcile(request)
+
+			err := k8sClient.Get(context.Background(), types.NamespacedName{
+				Namespace: hostingdeploymentautoscalingjob.ObjectMeta.Namespace,
+				Name:      hostingdeploymentautoscalingjob.ObjectMeta.Name,
+			}, hostingdeploymentautoscalingjob)
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(hostingdeploymentautoscalingjob.Status.HostingDeploymentAutoscalingJobStatus).To(Equal(FailedAutoscalingJobStatus))
```

Comment:
Hmm - Remind me to mention this, but we use "Error" for other controllers when there's been a 4xx/5xx response from SM.

Reply:
@RedbackThomson reminder

Reply:
Bit late now. This is fine.